auto-pos-creator/src/multi_contract_interactions/create_pos.rs

```diff
@@ -1,231 +1,202 @@
-pub enum StepsToPerform {
-    AddLiquidity,
-    EnterFarm,
-    EnterMetastaking,
-}
-
-pub struct CreatePosArgs<M: ManagedTypeApi> {
-    pub caller: ManagedAddress<M>,
-    pub dest_pair_address: ManagedAddress<M>,
-    pub pair_input_tokens: PairTokenPayments<M>,
-    pub steps: StepsToPerform,
-    pub first_token_min_amount_out: BigUint<M>,
-    pub second_token_min_amount_out: BigUint<M>,
-}
-
 #[multiversx_sc::module]
 pub trait CreatePosModule:
-    crate::external_sc_interactions::pair_actions::PairActionsModule
+    utils::UtilsModule
     + crate::configs::pairs_config::PairsConfigModule
-    + utils::UtilsModule
-    + auto_farm::whitelists::farms_whitelist::FarmsWhitelistModule
-    + auto_farm::whitelists::metastaking_whitelist::MetastakingWhitelistModule
-    + auto_farm::external_storage_read::farm_storage_read::FarmStorageReadModule
-    + auto_farm::external_storage_read::metastaking_storage_read::MetastakingStorageReadModule
+    + crate::external_sc_interactions::pair_actions::PairActionsModule
+    + crate::external_sc_interactions::egld_wrapper_actions::EgldWrapperActionsModule
+    + crate::external_sc_interactions::router_actions::RouterActionsModule
     + crate::external_sc_interactions::farm_actions::FarmActionsModule
     + crate::external_sc_interactions::metastaking_actions::MetastakingActionsModule
 {
-    fn create_pos_common(&self, args: CreatePosArgs<Self::Api>) -> PaymentsVec<Self::Api> {
-        let add_liq_result = self.call_pair_add_liquidity(
-            args.dest_pair_address,
-            args.pair_input_tokens.first_tokens,
-            args.pair_input_tokens.second_tokens,
-            args.first_token_min_amount_out,
-            args.second_token_min_amount_out,
-        );
-
-        let mut output_payments = PaymentsWrapper::new();
-        output_payments.push(add_liq_result.first_tokens_remaining);
-        output_payments.push(add_liq_result.second_tokens_remaining);
-
-        if matches!(args.steps, StepsToPerform::AddLiquidity) {
-            output_payments.push(add_liq_result.lp_tokens);
-
-            return output_payments.send_and_return(&args.caller);
-        }
-
-        let opt_enter_result = self.try_enter_farm_with_lp(&add_liq_result.lp_tokens, &args.caller);
-        require!(opt_enter_result.is_some(), COULD_NOT_CREATE_POS_ERR_MSG);
-
-        let enter_result = unsafe { opt_enter_result.unwrap_unchecked() };
-        output_payments.push(enter_result.rewards);
-
-        if matches!(args.steps, StepsToPerform::EnterFarm) {
-            output_payments.push(enter_result.new_farm_token);
-
-            return output_payments.send_and_return(&args.caller);
+    fn process_payment(
+        &self,
+        payment: EgldOrEsdtTokenPayment,
+        swap_operations: MultiValueEncoded<SwapOperationType<Self::Api>>,
+    ) -> EsdtTokenPayment {
+        let esdt_payment = self.get_esdt_payment(payment);
+        require!(esdt_payment.token_nonce == 0, "Only fungible ESDT accepted");
```

Comment:
`esdt_payment.token_nonce == 0` is also checked in the `get_esdt_payment` function.

Reply:
Removed.